C/eval.c

```diff
@@ -587,6 +597,35 @@ static bool runTCO(evalState state, call* stack, const dag_node* dag, type* type
     }
   }
   assert(pc == len);
+
+  return true;
+}
+
+/* Inspects the stack contents after a successful runTCO execution to verify anti-DOS properties:
+ * 1. All dag nodes were executed at least once.
+ * 2. Both branches of every CASE node were executed.
+ *
+ * If these are violoated it means that the dag should have had unused nodes prune or should have transformed
```

Comment:
In 2d58ec81f4a848b7f7083e3e85ba85e78545b7b6:

nit `violated`

Reply:
I would actually suggest rewriting this line, I find it hard to read. Maybe
```
If these are violated, it means that the dag had unused but unpruned nodes. For example, a CASE node may have two children, neither of which is an ASSERTL or ASSERTR node.
```

Reply:
Replaced the text with just something simple.